### ISSUE TITLE
No session registered supporting Version 1213486160

### ISSUE THREAD

**Issue by Arturski** [in3rsha/bitcoin-to-neo4j#3]
Hello,

Thanks for the awesome work! I have been working through the instructions and came across a problem starting the script.

```
arthur@cloud-ops1 ~/other-repos/bitcoin-to-neo4j $ php main.php
PHP Fatal error:  Uncaught InvalidArgumentException: No session registered supporting Version 1213486160 in /home/arthur/other-repos/bitcoin-to-neo4j/vendor/graphaware/neo4j-bolt/src/Protocol/SessionRegistry.php:88
Stack trace:
#0 /home/arthur/other-repos/bitcoin-to-neo4j/vendor/graphaware/neo4j-bolt/src/Driver.php(118): GraphAware\Bolt\Protocol\SessionRegistry->getSession(1213486160, Array)
#1 /home/arthur/other-repos/bitcoin-to-neo4j/vendor/graphaware/neo4j-php-client/src/Connection/Connection.php(192): GraphAware\Bolt\Driver->session()
#2 /home/arthur/other-repos/bitcoin-to-neo4j/vendor/graphaware/neo4j-php-client/src/Connection/Connection.php(109): GraphAware\Neo4j\Client\Connection\Connection->checkSession()
#3 /home/arthur/other-repos/bitcoin-to-neo4j/vendor/graphaware/neo4j-php-client/src/Client.php(67): GraphAware\Neo4j\Client\Connection\Connection->run('CREATE CONSTRAI...', NULL, NULL)
#4 /home/arthur/other-repos/bitcoin-to-neo4j/main.php(28): GraphAware\Neo4j\Client\Client->run('CREATE CONSTRAI...')
#5 {main}
  throw in /home/arthur/other-repos/bitcoin-to-neo4j/vendor/graphaware/neo4j-bolt/src/Protocol/SessionRegistry.php on line 88
```

I followed your instructions however I had to get a little creative with phpredis as the url in the README is not accessible. I managed to install the version below but I cant figure out what version it is complaining about. Is it the PHP neo4j driver?

```
arthur@cloud-ops1 /usr/local/share/phpredis-3.1.4 $ sudo make test

Build complete.
Don't forget to run 'make test'.


=====================================================================
PHP         : /usr/bin/php7.0 
PHP_SAPI    : cli
PHP_VERSION : 7.0.28-0ubuntu0.16.04.1
ZEND_VERSION: 3.0.0
PHP_OS      : Linux - Linux cloud-ops1 4.10.0-32-generic #36~16.04.1-Ubuntu SMP Wed Aug 9 09:19:02 UTC 2017 x86_64
INI actual  : /usr/local/share/phpredis-3.1.4/tmp-php.ini
More .INIs  :   
CWD         : /usr/local/share/phpredis-3.1.4
Extra dirs  : 
VALGRIND    : Not used
=====================================================================
TIME START 2018-03-21 19:56:02
=====================================================================
No tests were run.
arthur@cloud-ops1 /usr/local/share/phpredis-3.1.4 $ sudo make install
Installing shared extensions:     /usr/lib/php/20151012/

```
`neo4j version 3.3.4`

Thanks

**Comment by in3rsha** [comment]
I haven't been able to recreate the error myself, so I've sent a message to Chris Willemsen (the creator of the neo4j-php-client) on the Neo4j Slack channel (https://neo4j-users.slack.com).

**Comment by Arturski** [comment]
Thats awesome thanks, not my laptop at the moment will post the steps this weekend.

**Comment by ikwattro** [comment]
Hi @Arturski @in3rsha 

Can you try with Neo4j 3.3.2 and tell me if that solve your issue?

In the meantime, you can switch over to http protocol instead of boltt.

**Comment by ikwattro** [comment]
I just tested a 3.3.4 locally, no issue. I assume some configuration is wrong , like defined here : 

https://github.com/in3rsha/bitcoin-to-neo4j#config

**Comment by Arturski** [comment]
Thanks @ikwattro I did follow the config steps will double check and post back

@in3rsha I used the following steps to install phpredis

```
sudo wget https://github.com/phpredis/phpredis/archive/3.1.4.zip -O phpredis.zip
sudo unzip phpredis.zip
sudo rm phpredis.zip
cd phpredis-3.1.4
sudo ./configure
sudo make
sudo make install
```

You may want to double check these steps, i also ran the ./mkdeb script however I created my redis.ini in the php7 directory. Sorry I am not that experienced with PHP, hope it helps.

**Comment by in3rsha** [comment]
@Arturski Were you able to figure out what was causing this error? I haven't been able to replicate it myself.

Also, thanks for the notes on install phpredis.

**Comment by Arturski** [comment]
Hey Guys,

Yes I managed to get it working. The initial issue which I reported was actually caused by the fact that I stupidly changed the bolt port to the browser port.

Once i fixed that setting authentication kept failing with the default credentials, even after I used the neo4j password reset tool. 

```
PHP Fatal error:  Uncaught GraphAware\Bolt\Exception\MessageFailureException: Neo4j Exception "The client is unauthorized due to authentication failure." with code "Neo.ClientError.Security.Unauthorized" in /home/arthur/other-repos/bitcoin-to-neo4j/vendor/graphaware/neo4j-bolt/src/Protocol/V1/Session.php:263
Stack trace:
```
I resolved it by creating another admin user which worked perfectly. Thanks for your help guys and apologies for the false alarm.

@in3rsha I did notice a php warning when I first ran the script below, although i do have to add that i copied a couple of .dat files to a test directory just to get started so not sure if that had any impact.

```
Reading /home/arthur/other-repos/bitcoin-to-neo4j/test/blk00000.dat...

PHP Warning:  fseek() expects parameter 2 to be integer, string given in /home/arthur/other-repos/bitcoin-to-neo4j/main.php on line 73
```

**Comment by in3rsha** [comment]
Excellent. Glad to hear it, and thank you for documenting your steps so clearly.

I just fixed the PHP warning. When the script starts it checks to see if there a value set in Redis for a file pointer position (so that it can restart at the same place when you stop the script). I set it to default at 0 if it's not set, instead of just relying on false (which had the same effect anyway).

Also, this script takes an extremely long time to import the entire blockchain. It took me months. It's nice to have some local data to work on, but if you'd like to have access to a complete Neo4j database of the bitcoin blockchain:

http://learnmeabitcoin.com:7474/browser/
user: community
pass: bitcoingraph
